### PR TITLE
Enable encryption by default AWS instance EBS vols

### LINKED ISSUE #126: Enable encryption on disks attached to AWS VMs
**How to categorize this issue?**

/area security
/kind enhancement
/priority normal
/platform aws

**What would you like to be added**:
Encryption to be enabled by default for all disks attached to EC2 instances

**Why is this needed**:
Security

### PR DESCRIPTION
**Pull request by prashanth26**
**How to categorize this PR?**

/area security
/kind enhancement
/priority normal
/platform aws

**What this PR does / why we need it**:
The PR enables encryption by default for EBS volumes used by AWS EC2 instances. 

**Which issue(s) this PR fixes**:
Fixes #126 

**Special notes for your reviewer**:
This might cause issues with machines that do not support encryption by default. Refer list here - https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/EBSEncryption.html#ebs-encryption-requirements. 

However, this is something that can be overridden by setting the encryption flag to `false` on the [machine volume settings here](https://github.com/gardener/gardener/blob/master/example/90-shoot.yaml#L49). 

I checked with our cloud profile and I found 6 machine types with `g4dn` which don't have support for encryption. Something that we need to disable from the cloud profile?

**Release note**:

```noteworthy user
Enable encryption by default for AWS instance EBS volumes.
```
```action operator
Need to remove EC2 instances without encryption support in the cloud profile.
```

### CODE REVIEW

**Comment by timuthy** [comment]
> I checked with our cloud profile and I found 6 machine types with g4dn which don't have support for encryption. Something that we need to disable from the cloud profile?

The encrypted volumes feature was once [added](https://github.com/gardener/gardener/commit/6a9f80379ab1152b167d0d58914d1b42a7e48803) to the shoot API and if we want to exclude certain machine types, we'll need an analogue implementation for the CloudProfile + validation.

**Comment by hardikdr** [comment]
Changes via /lgtm.
Though need to follow up on if it breaks existing clusters - eg unsupported machine types.

**Comment by prashanth26** [comment]
> The encrypted volumes feature was once added to the shoot API and if we want to exclude certain machine types, we'll need an analogue implementation for the CloudProfile + validation.

Or we could retain the unsupported machines in the cloud profile as it is today. 

However, we could send out announcements mentioning shoot owners who would like to use machines without manual encryption support will have to update [their worker encryption flags] to false (https://github.com/gardener/gardener/blob/master/example/90-shoot.yaml#L49).

**Comment by timuthy** [comment]
> However, we could send out announcements mentioning shoot owners who would like to use machines without manual encryption support will have to update their worker encryption flags to false.

I'd rather go the cloud profile way to be sure. You can never reach any stakeholder by such an announcements, imo.

**Comment by rfranzke** [comment]
/assign @timuthy 
I get your point @timuthy, but given that not too many machine types would be affected I'd probably not block this PR and create an issue in g/g instead. For now, we can probably document that encryption is enabled by default and certain machine types don't support it (why it must be set to `false`).

**Comment by prashanth26** [comment]
> Please adapt https://github.com/gardener/gardener-extension-provider-aws/blob/master/docs/usage-as-end-user.md#workerconfig

Sorry missed that. I have updated the docs to reflect the latest state. PTAL.

**Comment by rfranzke** [comment]
Can you check the failing test?

**Comment by prashanth26** [comment]
I have fixed the tests.

**Comment by vpnachev** [comment]
@prashanth26 could you rephrase the second release note - I do not get it.
> Need to remove EC2 instances without encryption support in the cloud profile.

**Comment by prashanth26** [comment]
Hi @vpnachev ,

I have rewritten the release note. Hope it's better now.

**Comment by vpnachev** [comment]
Thanks! It is looking great now.

**Comment by vlerenc** [comment]
FYI: AWS reported that issues with encrypted disks can happen regardless of the disk size, which was new to me (reported were issues only with large disks: https://github.com/kubernetes/kubernetes/issues/86064). A call is scheduled for 21:00 Thursday (German time) to clarify that and more (like KCM/CSI handling of that case). Should we hold back the release of this component, at least until Friday or should we "risk" it @vpnachev @timuthy @rfranzke? On the other hand, different than KCM/CSI handled volume handling, we can in MCM handle the local disks always differently and not fail, right @prashanth26?

**Comment by prashanth26** [comment]
> On the other hand, different than KCM/CSI handled volume handling, we can in MCM handle the local disks always different and not fail, right @prashanth26?

The system disks (root & additional attached volumes) are encrypted by default using this PR. And this is agnostic of Kubernetes, as we are just attaching encrypted system disks now. I tried it out with a simple cluster, seems to work well. However, I am not sure about the size implication here. 

However, this is something that can be overridden by setting the encryption flag to `false` on the [machine volume settings here](https://github.com/gardener/gardener/blob/master/example/90-shoot.yaml#L49) in the worst case.

**Comment by vpnachev** [comment]
To be honest, we had not to change the default behavior at all. Encrypted volumes were available even before this PR.
But now it is in and In case of emergency, the encryption can be explicitly disabled by the shoot cluster owner, so I do not think this should be a blocker for the next release.

**Comment by rfranzke** [comment]
+1 @vpnachev
